### PR TITLE
Rename http_gui_port to app_route and add local config

### PR DESCRIPTION
**Pull request by guilhermedemouraa**

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -47,59 +55,75 @@
...
+    )
 
-@app.websocket("/subscribe/{service_name}/{uri_path}")
-async def subscribe(websocket: WebSocket, service_name: str, uri_path: str, every_n: int = 1):
+    all_uris = {}
+    for config in all_uris_list.configs:
+        if config.name == "all_subscription_uris":
+            for subscription in config.subscriptions:
+                uri = subscription.uri
+                service_name = uri.query.split("=")[1]
```
Comment (edgarriba) [inline main.py:68]: add a comment so that whoever understands why sply by '=' and the index position

File: main.py [hunk 2]
```diff
@@ -47,59 +55,75 @@
...
+    return JSONResponse(content=dict(sorted(all_uris.items())), status_code=200)
+
+
+@app.websocket("/subscribe/{service_name}/{uri_path:path}")
+@app.websocket("/subscribe/{service_name}/{sub_service_name}/{uri_path:path}")
+async def subscribe(
+    websocket: WebSocket,
+    service_name: str,
+    uri_path: str,
+    sub_service_name: str = None,
```
Comment (edgarriba) [inline main.py]:
```suggestion
    sub_service_name: Optional[str] = None,
```

Comment (edgarriba) [reply main.py]: None 's are typed with Optional

File: main.py [hunk 3]
```diff
@@ -47,59 +55,75 @@
...
+    every_n: int = 1
+):
     """Coroutine to subscribe to an event service via websocket.
     
     Args:
         websocket (WebSocket): the websocket connection
         service_name (str): the name of the event service
+        sub_service_name (str, optional): the sub service name, if any
```
Comment (edgarriba) [inline main.py]: i would keep same order as in the signature

File: main.py [hunk 4]
```diff
@@ -47,59 +55,75 @@
...
         every_n (int, optional): the frequency to receive events. Defaults to 1.
     
     Usage:
-        ws = new WebSocket("ws://localhost:8042/subscribe/oak0/left
+        ws = new WebSocket("ws://localhost:8042/subscribe/gps/pvt")
+        ws = new WebSocket("ws://localhost:8042/subscribe/oak/0/imu")
     """
 
-    client: EventClient = clients[service_name]
+    if sub_service_name:
```
Comment (edgarriba) [inline main.py]: for simplicity you could reduce to a single if else statement branch that set a single variable and have single client creation line

**Review by edgarriba — APPROVED** [review]

File: main.py (hunk 1 above)
Comment (guilhermedemouraa) [reply main.py:68]: Good point! Addressed in commit 80ddebdcd28e5406ab03b4feeb258033ebc18f47

File: main.py (hunk 2 above)
Comment (guilhermedemouraa) [reply main.py]: Good point! Addressed in commit 80ddebdcd28e5406ab03b4feeb258033ebc18f47

File: main.py (hunk 4 above)
Comment (guilhermedemouraa) [reply main.py]: Agreed! Addressed in commit 80ddebdcd28e5406ab03b4feeb258033ebc18f47

**Pull request merged**